compute/do-droplets.js

```diff
@@ -0,0 +1,169 @@
+class Droplets {
+  /**
+   * Droplets constructor
+   * @constructor
+   * @param {object} doSdk - DigitalOcean SDK
+   * @param {string} token - DigitalOcean token
+   */
+  constructor(doSdk, token) {
+    this._DO = doSdk;
+    this._client = new this._DO.client(token);
+  }
+
+  /**
+   * List all Droplets
+   * @list
+   */
+  list() {
+    return new Promise((resolve,reject) => {
+      this._client.droplets.list((err,result) => {
+        if (err) reject(err)
+        console.log(result)
+        resolve(result)
+      })
+    });
+  }
+
+  /**
+   * Retrieve an existing Droplet by id
+   * @listByID
+   * @param {string} id
+   */
+  listByID(id) {
+    return new Promise((resolve,reject) => {
+      this._client.droplets.get(id,(err,result) => {
+        if (err) reject(err)
+        console.log(result)
+        resolve(result)
+      })
+    });
+  }
+
+  /**
+   * Create a Droplet
+   * @create
+   * @param {object} params
+   */
+  create(params) {
+    return new Promise((resolve,reject) => {
+      this._client.droplets.create(params,(err,result) => {
+        if (err) reject(err)
+        if (result.length > 1) {
+          var ids = []
+          for (var i=0; i<result.length; i++) {
+            ids.push(result[i].id)
+          }
+          resolve("Created new droplets with ids: " + ids)
+        }
+        else {
+          resolve("Created a new droplet with id: " + result.id)
+        }
+      })
+    });
+  }
+
+  /**
+   * List all available Kernels for a Droplet
+   * @listKernels
+   * @param {string} id
+   */
+  listKernels(id) {
```

Comment:
What's the equivalent function in aws-plugin?

Reply:
There's a way to connect a pre-created Linux instance using SSH. But that can be achieved only using a manual series of steps and not through a delegated API or SDK, which is in contrast to what DigitalOcean provides.

TL;DR We can't achieve the same using aws-plugin